homeassistant/components/remember_the_milk/__init__.py

```diff
@@ -160,56 +160,64 @@ class RememberTheMilkConfiguration:
     This class stores the authentication token it get from the backend.
     """
 
-    def __init__(self, hass):
+    def __init__(self, hass: HomeAssistant) -> None:
         """Create new instance of configuration."""
         self._config_file_path = hass.config.path(CONFIG_FILE_NAME)
-        if not os.path.isfile(self._config_file_path):
-            self._config = {}
-            return
+        self._config = {}
+        _LOGGER.debug("Loading configuration from file: %s", self._config_file_path)
         try:
-            _LOGGER.debug("Loading configuration from file: %s", self._config_file_path)
-            with open(self._config_file_path, encoding="utf8") as config_file:
-                self._config = json.load(config_file)
+            self._config = json.loads(
+                Path(self._config_file_path).read_text(encoding="utf8")
+            )
+        except FileNotFoundError:
+            _LOGGER.debug("Missing configuration file: %s", self._config_file_path)
+        except OSError:
+            _LOGGER.debug(
+                "Failed to read from configuration file, %s, using empty configuration",
+                self._config_file_path,
+            )
         except ValueError:
             _LOGGER.error(
-                "Failed to load configuration file, creating a new one: %s",
+                "Failed to parse configuration file, %s, using empty configuration",
                 self._config_file_path,
             )
```

Comment:
Is this something we ideally should split out in its separate function as this is a side effect for the constructor right?

Reply:
Yes, but I didn't want to change more in this PR as it was mostly aimed at typing and tests. I'll follow up with more changes.